app/page.tsx

```diff
@@ -1,17 +1,35 @@
-'use client';
+'use client'
 
+import EventTable from '@/components/EventTable';
+import RouteButton from '@/components/RouteButton';
 import { UserButton } from '@clerk/nextjs'
-import { useRouter } from 'next/navigation'
+import { useEffect, useState } from 'react';
+import { Event } from '@prisma/client';
 
 export default function Home() {
-	const router = useRouter();
+	const [events, setEvents] = useState<Event[]>([]);
+
+	useEffect(() => {
+		const getEvents = async () => {
```

Comment:
Since we are running this function in multiple places we should make it a custom hook to avoid code repetition.

Reply:
So `useEffect` in this file gets **all events**, but in `manageEvent/[id]/page.tsx`, it gets an **event by id** to display on that page. So they're not actually the same.

But do you mean we should just create a hook for getting all events, and then in `manageEvent/[id]/page.tsx`, we use that hook and loop for the specific event by id with client-side code?

Reply:
Oh my bad I thought they did the same thing, we could leave it as is or we could make a hook that takes in an optional parameter of id and if that parameter is passed in we can search by id instead of getting all of them. It's up to you to decide what you want to do, I think both approaches are fine for the time being. 

I think getting all of the events and looping through each event to find the one with the corresponding id is not an optimized approach.

Reply:
ig we can just leave it as is for now, but if it gets to a point where we gotta repeat code then yea we'll just create a hook for that, so good looking out